# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: d621668762f4
# Full Hash: d621668762f41d6c8747dda9b35382ab007b75b0
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Move AudioConduit to call worker thread for mCall and related access. r=ng,bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112852
# ==============================================================================

diff -r 1c3926cd7004 -r d621668762f4 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Apr 07 11:36:28 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Apr 07 15:01:03 2021 +0200
@@ -206,20 +206,22 @@
   // Accessed on any thread under mTransportMonitor.
   RefPtr<TransportInterface> mReceiverTransport;
 
-  // Const so can be accessed on any thread. Most methods are called on
-  // main thread.
+  // Const so can be accessed on any thread. Most methods are called on the Call
+  // thread.
   const RefPtr<WebrtcCallWrapper> mCall;
 
-  // Accessed only on main thread.
+  // Accessed only on the Call thread.
   webrtc::AudioReceiveStream::Config mRecvStreamConfig;
 
-  // Written only on main thread. Guarded by mMutex, except for reads on main.
+  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Call thread.
   webrtc::AudioReceiveStream* mRecvStream;
 
-  // Accessed only on main thread.
+  // Accessed only on the Call thread.
   webrtc::AudioSendStream::Config mSendStreamConfig;
 
-  // Written only on main thread. Guarded by mMutex, except for reads on main.
+  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Call thread.
   webrtc::AudioSendStream* mSendStream;
 
   // accessed on creation, and when receiving packets
@@ -229,19 +231,24 @@
   RtpPacketQueue mRtpPacketQueue;
 
   // If true => mSendStream started and not stopped
-  // Written only on main thread. Guarded by mMutex, except for reads on main.
+  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Call thread.
   bool mSendStreamRunning;
   // If true => mRecvStream started and not stopped
-  // Written only on main thread. Guarded by mMutex, except for reads on main.
+  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Call thread.
   bool mRecvStreamRunning;
 
-  // Accessed only on main thread.
+  // Accessed only on the Call thread.
   bool mDtmfEnabled;
   int mDtmfPayloadType = -1;
   int mDtmfPayloadFrequency = -1;
 
   Mutex mMutex;
 
+  // Call worker thread. All access to mCall->Call() happens here.
+  const RefPtr<AbstractThread> mCallThread;
+
   // Socket transport service thread. Any thread.
   const nsCOMPtr<nsISerialEventTarget> mStsThread;
 
